arangod/IResearch/IResearchCalculationAnalyzer.cpp

```diff
@@ -0,0 +1,507 @@
+  //              paramRef == arangodb::velocypack::StringRef(c->getStringValue(),
+  //                c->getStringLength())) {
+  //              c = newNode;
+  //              // no break here. replace all occurrences
+  //            }
+  //          }
+  //        }
+  //      }
+  //      node = newNode;
+  //    }
+  //  } else if (node->type == NODE_TYPE_BOUND_ATTRIBUTE_ACCESS) {
+  //    // look at second sub-node. this is the (replaced) bind parameter
+  //    auto name = node->getMember(1);
+
+  //    if (name->type == NODE_TYPE_VALUE) {
+  //      if (name->value.type == VALUE_TYPE_STRING && name->value.length != 0) {
+  //        // convert into a regular attribute access node to simplify handling
+  //        // later
+  //        return createNodeAttributeAccess(node->getMember(0), name->getStringValue(),
+  //          name->getStringLength());
+  //      }
+  //    } else if (name->type == NODE_TYPE_ARRAY) {
+  //      // bind parameter is an array (e.g. ["a", "b", "c"]. now build the
+  //      // attribute accesses for the array members recursively
+  //      size_t const n = name->numMembers();
+
+  //      AstNode* result = nullptr;
+  //      if (n > 0) {
+  //        result = node->getMember(0);
+  //      }
+
+  //      for (size_t i = 0; i < n; ++i) {
+  //        auto part = name->getMember(i);
+  //        if (part->value.type != VALUE_TYPE_STRING || part->value.length == 0) {
+  //          // invalid attribute name part
+  //          result = nullptr;
+  //          break;
+  //        }
+
+  //        result = createNodeAttributeAccess(result, part->getStringValue(),
+  //          part->getStringLength());
+  //      }
+
+  //      if (result != nullptr) {
+  //        return result;
+  //      }
+  //    }
+  //    // fallthrough to exception
+
+  //    // if no string value was inserted for the parameter name, this is an
+  //    // error
+  //    THROW_ARANGO_EXCEPTION_PARAMS(TRI_ERROR_QUERY_BIND_PARAMETER_TYPE,
+  //      node->getString().c_str());
+  //  } else if (node->type == NODE_TYPE_TRAVERSAL) {
+  //    extractCollectionsFromGraph(node->getMember(2));
+  //  } else if (node->type == NODE_TYPE_SHORTEST_PATH ||
+  //    node->type == NODE_TYPE_K_SHORTEST_PATHS) {
+  //    extractCollectionsFromGraph(node->getMember(3));
+  //  }
+
+  //  return node;
+  //};
+  //Ast::traverseAndModify(_ast->root(), func);
+}
+
+bool CalculationAnalyzer::next() {
+  if (_queryResults.get()) {
+    // TODO: call engine refill!
+    while (_queryResults->numRows() > _resultRowIdx) {
+      AqlValue const& value = _queryResults->getValueReference(_resultRowIdx++, _engine->resultRegister());
+      if (value.isString()) {
+        arangodb::velocypack::Builder builder;
+        value.toVelocyPack(&arangodb::velocypack::Options::Defaults, builder, true, false);
+        _str = builder.slice().copyString();
+        _term.value = irs::ref_cast<irs::byte_type>(irs::string_ref(_str));
+        return true;
+      }
```

Comment:
`#include "IResearch/AqlHelper.h"`
+
```suggestion
if (value.isString()) {
  _term.value = arangodb::iresearch::getStringRef(value);
}
```